### PR TITLE
-st-pagination (DO NOT MERGE YET)

### PR DESCRIPTION
**Pull request by JoMarton**
Converting Pagination component from JSS to stylable.

### CODE REVIEW

File: packages/wix-ui-core/src/components/Pagination/Pagination.st.css [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+.pageForm {
+    display: flex;
+    order: 3;
+}
+
+.pageInput {}
+.totalPages {}
+.slash {}
+.disabled {}
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Pagination/Pagination.st.css]: Don't need `.disabled` anymore

File: packages/wix-ui-core/src/components/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -64,19 +35,19 @@ export interface PaginationProps {
   showInputModeTotalPages?: boolean;
   responsive?: boolean;
   maxPagesToShow?: number;
-  classes?: PaginationClasses;
   id?: string;
   updateResponsiveLayout?: (callback: () => void) => void;
+  style?: React.CSSProperties;
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Pagination/Pagination.tsx]: Do you plan to remove props.width later?

File: packages/wix-ui-core/src/components/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -214,51 +184,51 @@ class Pagination extends React.Component<PaginationProps, PaginationState> {
...
+      [ButtonType.First]: [pStyle.navButtonFirst, this.props.firstLabel, 1],
+      [ButtonType.Last]: [pStyle.navButtonLast, this.props.lastLabel, totalPages]
     }[type] as [string, string, number];
 
     return (
       <a
         data-hook={type}
         id={this.getId('navButton' + upperCaseFirst(type))}
-        className={classNames(classes.navButton, btnClass, {[classes.disabled]: disabled})}
+        {...pStyle('', {disabled})}
```
Comment (alisey) [inline packages/wix-ui-core/src/components/Pagination/Pagination.tsx]: I wonder what Barak says about this, maybe there's another way.

File: packages/wix-ui-core/stories/Pagination/pagination-story-theme.st.css [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+.root {
+    -st-extends: Pagination;
+    color: value(materialBlueGray700);
+    background-color: #ECEFF1;
+
+}
+
+.root::navButton {
+    -st-mixin: button-common-mixin;
+    background-color: value(materialBlueGray100);;
```
Comment (alisey) [inline packages/wix-ui-core/stories/Pagination/pagination-story-theme.st.css]: double semicolon

File: packages/wix-ui-core/stories/Pagination/pagination-story.tsx [hunk 5]
```diff
@@ -1,27 +1,30 @@
...
-import {theme} from './pagination-story-theme';
+import {Pagination as CorePagination, PaginationProps} from '../../src/components/Pagination';
+import style from './pagination-story-theme.st.css';
+import {withStylable} from '../../src/utils/withStylable';
 
 // Assuming we use pagination-story-theme
 const spaceForPages = (n) => (n + 2) * 40;
 
-class UncontrolledPagination extends Pagination {
+const Pagination = withStylable<PaginationProps, {}>(CorePagination, style , () => ({}));
```
Comment (alisey) [inline packages/wix-ui-core/stories/Pagination/pagination-story.tsx]: space before comma 😱

File: packages/wix-ui-core/src/components/Pagination/Pagination.tsx (hunk 3 above)
Comment (JoMarton) [reply packages/wix-ui-core/src/components/Pagination/Pagination.tsx]: ok, so we decided to use `pStyle('navButton btnClass', {disabled})`

File: packages/wix-ui-core/src/components/Pagination/Pagination.tsx (hunk 2 above)
Comment (JoMarton) [reply packages/wix-ui-core/src/components/Pagination/Pagination.tsx]: probably, need to discuss the use cases

File: packages/wix-ui-core/src/components/Pagination/PageStrip.tsx [hunk 6]
```diff
@@ -78,14 +69,14 @@ export class PageStrip extends React.Component<PageStripProps, PageStripState> {
...
+        className={style.pageStrip}
       >
-        <div className={this.props.classes.pageStripInner}>
+        <div className={style.pageStripInner}>
           {this.renderLayout(this.getLayout(), false)}
         </div>
 
         {this.isResponsive() &&
-          <div className={this.props.classes.pageStripInner + ' ' + this.props.classes.pageStripTemplate}>
+          <div className={style.pageStripInner + ' ' + style.pageStripTemplate}>
```
Comment (amiryonatan) [inline packages/wix-ui-core/src/components/Pagination/PageStrip.tsx]: `${style.pageStripInner} ${style.pageStripTemplate}`

**Pull request merged**